src/snowflake/snowpark/modin/plugin/extensions/dataframe_overrides.py

```diff
@@ -382,10 +382,9 @@ def __delitem__(self, key):
 
 
 @register_dataframe_accessor("style")
-@dataframe_not_implemented()
 @property
 def style(self):  # noqa: RT01, D200
-    pass  # pragma: no cover
+    return self._to_pandas().style
```

Comment:
I think materializing is fine for now, but if performance becomes a concern we should consider returning a custom object in the future. A styler can be configured to hide rows/columns and filter data, which we would be able to do before any materializations.

Reply:
Agreed. There could be opportunities to push some of the styler post processing down to the snowflake engine. But I'd defer this for now - until there is a clear signal from customers that this is high priority. I think the current implementation handles the most common use cases. Note that the [docs](https://github.com/pandas-dev/pandas/blob/68d9dcab5b543adb3bfe5b83563c61a9b8afae77/doc/source/user_guide/style.ipynb#L4) say:
> The Styler, which can be used for large data but is primarily designed for small data, currently has the ability to output to these formats